Auth and email generator pradeep

PRADEEPERIYASAMY opened the pull request.

PRADEEPPERIYASAMYTAMU committed:
db: Create users table schema

PRADEEPPERIYASAMYTAMU committed:
feat: Add User model with validations and associations

PRADEEPPERIYASAMYTAMU committed:
feat: Add initial routes for users and sessions

PRADEEPPERIYASAMYTAMU committed:
feat: Implement JWT authentication helpers in ApplicationController

PRADEEPPERIYASAMYTAMU committed:
feat: Add UsersController for signup and profile management

PRADEEPPERIYASAMYTAMU committed:
feat: Add SessionsController for login and logout

PRADEEPPERIYASAMYTAMU committed:
db: Create templates table schema

PRADEEPPERIYASAMYTAMU committed:
feat: Add Template model with associations

PRADEEPPERIYASAMYTAMU committed:
feat: Add TemplatesController with CRUD actions

PRADEEPPERIYASAMYTAMU committed:
db: Create automations table schema

PRADEEPPERIYASAMYTAMU committed:
feat: Add Automation model with scopes

PRADEEPPERIYASAMYTAMU committed:
feat: Create main application layout

PRADEEPPERIYASAMYTAMU committed:
feat: Create index view for templates

PRADEEPPERIYASAMYTAMU committed:
feat: Create shared form partial for templates

PRADEEPPERIYASAMYTAMU committed:
feat: Create new template view

PRADEEPPERIYASAMYTAMU committed:
feat: Create edit template view

PRADEEPPERIYASAMYTAMU committed:
chore: Configure import map for Stimulus and libraries

PRADEEPPERIYASAMYTAMU committed:
feat(stimulus): Add controller for spreadsheet verification

PRADEEPPERIYASAMYTAMU committed:
feat(stimulus): Add controller for dynamic rule editor UI

PRADEEPPERIYASAMYTAMU committed:
feat: Add Turbo Stream view for preview updates

PRADEEPPERIYASAMYTAMU committed:
feat: Create partial for displaying preview results

PRADEEPPERIYASAMYTAMU committed:
feat: Create partial for displaying preview errors

PRADEEPPERIYASAMYTAMU committed:
feat: added user auth views

PRADEEPPERIYASAMYTAMU committed:
feat: added shared code

PRADEEPPERIYASAMYTAMU committed:
fix: updated gemfile

ajitrskul committed:
Merge pull request #30 from TAMUCSCE-606-Zapmail/auth-and-email-generator-pradeep

Auth and email generator pradeep

Commit:
feat: Add automations association to User model

Commit:
feat: Add unit test on user model

Install rspec, add the unit test for user model and pass all test cases

Commit:
feat: Add scopes and helper methods to Automation model with tests

Unit test partially fail, need to be fix

Commit:
fix: Correct Automation model test validations and ordering

Issue 1: Status validation test failing
The test expects the automation to be invalid without status, but it's getting a default value. We need to explicitly set status: nil.

Issue 2: Recent scope ordering
The test expects ordering by send_at descending, but automations created at nearly the same time have the same send_at. We need to make the times more distinct.

Commit:
feat: Add automations_controller, unit test

pass partial tests

Commit:
fix: Add rails-controller-testing to test controller behavior directly

rails 5 remove assign(:variable) and render_template(:index), adding rails-controller-testing make these two methods work

Commit:
feat: add automation user interface

PRADEEPERIYASAMY committed:
Merge pull request #32 from TAMUCSCE-606-Zapmail/view_automation_history_v2

Automation History Page

pwzerus committed:
Add user profile edit tests and validations

pwzerus committed:
Delete OneDrive/Desktop/MIS/MIS第三学期/software engineering/Software-zapmail directory

pwzerus committed:
Overwrite files with updated ZapMail code

pwzerus committed:
Merge branch '21-edit-user-profile' of https://github.com/TAMUCSCE-606-Zapmail/ZapMail into 21-edit-user-profile

Commit:
db: Create mock user data in seeds.rb

ajitrskul committed:
Add jetbrains files to gitignore

Updated gitignore with jetbrains files so local application files are not added to application. Also bundle installed again.

ajitrskul committed:
Merge remote-tracking branch 'origin/auth-and-email-generator-pradeep' into view-current=strategies-and-rules-v2

Pulled changes from Pradeep's branch to get latest working version.

Commit:
feat: add pagination to automation history page

ajitrskul committed:
Refactored UI

Changed the colors and layout of headers and template editting page for better aesthetics, added logo to header, and rearragned header layout.

Commit:
fix: Remove wrong data in seeds.rb

ajitrskul committed:
Configured Cucumber and SimpleCov

Download and configured basic cucumber and simplecov features so we can check test coverage.

Commit:
feat: add automation history detail view

pwzerus committed:
Merge branch 'auth-and-email-generator-pradeep' into 21-edit-user-profile

PRADEEPERIYASAMY committed:
Merge pull request #34 from TAMUCSCE-606-Zapmail/view-automation-page-features

View automation history page features

pwzerus committed:
Merge branch 'auth-and-email-generator-pradeep' into 21-edit-user-profile

ajitrskul committed:
Fixed Subject DropDown Styling

Fixed subject dropdown styling to no longer be transparent and follow look of the remaining application.

PRADEEPERIYASAMY committed:
Merge pull request #33 from TAMUCSCE-606-Zapmail/21-edit-user-profile

21 edit user profile

PRADEEPPERIYASAMYTAMU committed:
feat: add job schedulers

PRADEEPPERIYASAMYTAMU committed:
fix: merge issue

PRADEEPPERIYASAMYTAMU committed:
fix: ui rendering issue

pinkpig777 reviewed (approved):
All major features have been implemented and verified to be functioning as intended. The codebase has undergone prior reviews before this major pull request, ensuring consistency with project standards and reducing the likelihood of regressions.

This PR represents the first deployable iteration of the application, which will be released on Heroku for further evaluation. Deployment will allow us to validate the system in a live environment, gather feedback, and prepare for subsequent improvements in later iterations.

ajitrskul committed:
Merge remote-tracking branch 'origin/auth-and-email-generator-pradeep' into view-current=strategies-and-rules-v2

Resolved merge conflicts between branches to pull request.

PRADEEPERIYASAMY committed:
Merge pull request #37 from TAMUCSCE-606-Zapmail/view-current-strategies-and-rules-v2

Merge view current strategies and rules v2 with primary feature branch

PRADEEPPERIYASAMYTAMU committed:
fix: refactor ui to have common theme

PRADEEPPERIYASAMYTAMU committed:
Restore zapmail logo image from remote branch

PRADEEPPERIYASAMYTAMU committed:
feat: merge issue

ajitrskul committed:
completed template page styling

Changed templates to drop index page to be a list view with drop downs that show further detail about each template which the user can edit with the existing edit button.

ajitrskul committed:
added home page

Created new controller and view for home page and changed app root and nav bar to reflect that

PRADEEPERIYASAMY committed:
Merge pull request #38 from TAMUCSCE-606-Zapmail/view-current-strategies-and-rules-v2

View current strategies and rules v2

PRADEEPPERIYASAMYTAMU committed:
fix: fix index page ui to resemble the theme in other pages

PRADEEPPERIYASAMYTAMU committed:
fix: resolve merge conflict

PRADEEPPERIYASAMYTAMU committed:
feat: added pagination to tempaltes and preview page

PRADEEPPERIYASAMYTAMU committed:
fix: remove the repeated scheduling logic from the email cron job handler

PRADEEPPERIYASAMYTAMU committed:
Configure application for Heroku deployment

PRADEEPPERIYASAMYTAMU committed:
feat: Configure Gemfile groups for production

PRADEEPPERIYASAMYTAMU committed:
feat: Configure Gemfile groups for production

PRADEEPPERIYASAMYTAMU committed:
fix: Align Ruby version and add Linux platform for Heroku

PRADEEPPERIYASAMYTAMU committed:
fix: Align Ruby version and add Linux platform for Heroku

PRADEEPPERIYASAMYTAMU committed:
chore: Skip asset compilation on Heroku

PRADEEPPERIYASAMYTAMU committed:
fix: Move tailwindcss-rails to development group

PRADEEPPERIYASAMYTAMU committed:
fix: Move dotenv-rails to main group for production

PRADEEPPERIYASAMYTAMU committed:
fix: Configure ActiveJob and ActionMailer for production

PRADEEPPERIYASAMYTAMU committed:
fix: Remove outdated assets.rb initializer

PRADEEPPERIYASAMYTAMU committed:
chore: Re-enable asset pipeline on Heroku

PRADEEPPERIYASAMYTAMU committed:
fix: Add executable build script for Heroku importmaps

PRADEEPPERIYASAMYTAMU committed:
fix: Remove tailwindcss-rails gem to resolve build conflicts

PRADEEPPERIYASAMYTAMU committed:
fix: Resolve build conflicts

PRADEEPPERIYASAMYTAMU committed:
feat: Add Slack monitoring for background jobs

PRADEEPPERIYASAMYTAMU committed:
fix: Update Gemfile.lock for slack-notifier

PRADEEPPERIYASAMYTAMU committed:
feat: add slack notification and updates

ajitrskul committed:
Fixed Local Server Hosting Issue

Readded sprockets and manifest.js with correctly pre-compiled assets to allow for local hosting in development

ajitrskul committed:
Deleted duplicate automation code

Deleted duplication automation class code that was clearly being overwritten by a more detailed version.

ajitrskul committed:
Fixed home page animations

Reconfigured home page animations to work with hotwired stimulus and function as desired.

ajitrskul committed:
Fixed addition home page CSS and styling issues

Re-added css files and styles to fix tamu home page css not rendering correctly and fixed additional home page animation and coloring issues.

ajitrskul committed:
Fixed spec indentation error

Refactored indents in user_spec due to error when running rspec and simplecov

ajitrskul committed:
Update README

updated readme to previous state from nuked branch

ajitrskul committed:
Updated Readme (again)

changed small readme typo/error from previous iteration.

ajitrskul committed:
Fixed transparent dropdown bug

Added back tribute container classes to fix transparent dropdown bug on edit template section.

PRADEEPPERIYASAMYTAMU committed:
fix: replace automation mailer code

PRADEEPPERIYASAMYTAMU committed:
fix: mounting the process tracker and email tracker for prod to trace issues

PRADEEPPERIYASAMYTAMU committed:
fix: mounting the process tracker and email tracker for prod to trace issues

PRADEEPPERIYASAMYTAMU committed:
fix: unmounting the process tracker and email tracker for prod to trace issues

PRADEEPPERIYASAMYTAMU committed:
fix: update the production rb code for hosting

PRADEEPPERIYASAMYTAMU committed:
fix: udpate redis config for prod

PRADEEPPERIYASAMYTAMU committed:
fix: udpate redis config for prod

PRADEEPPERIYASAMYTAMU committed:
fix: udpate redis config for prod

PRADEEPPERIYASAMYTAMU committed:
fix: udpate redis config for prod

PRADEEPPERIYASAMYTAMU committed:
fix: udpate redis config for prod

PRADEEPPERIYASAMYTAMU committed:
fix: udpate redis config for prod

PRADEEPPERIYASAMYTAMU committed:
fix: udpated automation table rendering issue

PRADEEPPERIYASAMYTAMU committed:
fix: udpated sent to completed

ajitrskul committed:
Merge remote-tracking branch 'origin/auth-and-email-generator-pradeep' into view-current-strategies-and-rules-v2

PRADEEPPERIYASAMYTAMU committed:
fix: added authorization for template controller

ajitrskul committed:
Deleted console.log and fixed slack notif config

Deleted the console.log  message in home animation.js and added if condition in slack.rb to ensure that application can still be run without slackwebhook.

ajitrskul committed:
Changed readme and sprocket config

Minor change to readme for new deployment link and added production env sprocket change

PRADEEPPERIYASAMYTAMU committed:
fix: update user fix

ajitrskul committed:
Fixed home page svg background shape

Changed svg background shape on home page to intended texas svg as it was unintentionally some abstract art

PRADEEPPERIYASAMYTAMU committed:
fix: updated turbo redering condition to have page visibility in all cases

ajitrskul committed:
Fixed automation controller specs

Fixed automation controller specs to function as desired due to changes in controller that made tests fail.

ajitrskul committed:
FIx automation spec & delete duplicate code

Deleted duplicate code in users controller and commented out some failing tests to fix in next couple commits.

ajitrskul committed:
Merge remote-tracking branch 'origin/auth-and-email-generator-pradeep' into view-current-strategies-and-rules-v2

PRADEEPPERIYASAMYTAMU committed:
Merge pull request #41 from TAMUCSCE-606-Zapmail/view-current-strategies-and-rules-v2

Added Home Page Styles & Fixed UI

PRADEEPPERIYASAMYTAMU committed:
fix: merge conflict

PRADEEPPERIYASAMYTAMU committed:
fix: powershaft issue with sprockets-rails

PRADEEPPERIYASAMYTAMU committed:
fix: added logo to the brwoser tab
